diff --git a/forward_authentication_service/libs/dnsconfig.sh b/forward_authentication_service/libs/dnsconfig.sh
index 35da1e0..d63731c 100755
--- a/forward_authentication_service/libs/dnsconfig.sh
+++ b/forward_authentication_service/libs/dnsconfig.sh
@@ -55,22 +55,14 @@ delete_114s() {
 }

 restart_dnsmasq() {
-	if [ "$uciconfig" = "" ]; then
-		systemctl restart dnsmasq
-	else
-		service dnsmasq restart
-	fi
+	kill -9 $(pidof dnsmasq)
+	dnsmasq -C /etc/dnsmasq.conf
 }

 reload_dnsmasq() {
-	if [ "$uciconfig" = "" ]; then
-		systemctl reload dnsmasq
-	else
-		service dnsmasq reload
-	fi
+	restart_dnsmasq
 }

-
 if [ "$setconf" = "" ]; then
 	exit 1

@@ -101,7 +93,8 @@ elif [ "$setconf" = "hostconf" ]; then
 		# Generic Linux
 		host_entry="$gw_ip $gw_fqdn"
 		# generate a tmp filename on tmpfs
-		ram_hosts=$(mktemp --tmpdir=/run/tmpfiles.d)
+		mkdir /run/tmpfiles.d 2>/dev/null
+		ram_hosts=$(mktemp -p /run/tmpfiles.d)

 		cp -p /etc/hosts "$ram_hosts" &&
 		(

diff --git a/forward_authentication_service/libs/libopennds.sh b/forward_authentication_service/libs/libopennds.sh
index ae3a7d1..59024b0 100755
--- a/forward_authentication_service/libs/libopennds.sh
+++ b/forward_authentication_service/libs/libopennds.sh
@@ -1017,11 +1017,8 @@ check_mhd() {
 		debugtype="warn"
 		write_to_syslog

-		if [ "$uci" -eq 1 ]; then
-			systemctl restart opennds
-		else
-			/etc/init.d/opennds restart
-		fi
+		kill -9 $(pidof opennds)
+		opennds -f

 		exit 0
 	fi
@@ -1296,11 +1293,7 @@ pre_setup () {

 		uci=$(type uci)

-		if [ -z "$uci" ]; then
-			systemctl stop opennds
-		else
-			/etc/init.d/opennds stop
-		fi
+		kill -9 $(pidof opennds)

 		exit 1
