diff --git a/src/main.c b/src/main.c
index c39fff6..8b4f3a9 100644
--- a/src/main.c
+++ b/src/main.c
@@ -353,31 +353,6 @@ setup_from_config(void)

 	debug(LOG_NOTICE, "MHD version is %s", version);

-	if (sscanf(version, "%d.%d.%d", &major, &minor, &patch) == 3) {
-
-		if (major < MIN_MHD_MAJOR) {
-			outdated = 1;
-
-		} else if (minor < MIN_MHD_MINOR) {
-			outdated = 1;
-
-		} else if (patch < MIN_MHD_PATCH) {
-			outdated = 1;
-		}
-
-		if (outdated == 1) {
-			debug(LOG_ERR, "libmicrohttpd is out of date, please upgrade to version %d.%d.%d or higher",
-				MIN_MHD_MAJOR, MIN_MHD_MINOR, MIN_MHD_PATCH);
-
-			if (config->use_outdated_mhd == 0) {
-				debug(LOG_ERR, "exiting...");
-				exit(1);
-			} else {
-				debug(LOG_ERR, "Attempting use of outdated MHD - Data may be corrupted or openNDS may fail...");
-			}
-		}
-	}
-
 	// Check routing configuration
 	int watchdog = 0;
 	int routercheck;
